app/components/WorkoutExercise/WorkoutExerciseRecordsView/WorkoutExerciseRecordsView.tsx

```diff
@@ -1,23 +1,28 @@
 import { observer } from 'mobx-react-lite'
-import { getParentOfType } from 'mobx-state-tree'
-import React from 'react'
+import React, { useMemo } from 'react'
 import { View, ScrollView } from 'react-native'
 
 import EmptyState from 'app/components/EmptyState'
 import { useStores } from 'app/db/helpers/useStores'
-import { WorkoutModel, WorkoutSet } from 'app/db/models'
 import { translate } from 'app/i18n'
 import { navigate } from 'app/navigators'
-import RecordsListItem from './RecordsListItem'
 import { PressableHighlight } from 'designSystem'
+import RecordsListItem from './RecordsListItem'
+import { WorkoutSet } from 'app/db/models'
 
 const WorkoutExerciseRecordsView: React.FC = () => {
-  const { openedExerciseRecords, stateStore } = useStores()
+  const { stateStore } = useStores()
+
+  const recordSets = useMemo(() => {
+    const openedExerciseRecords = stateStore.getOpenedExerciseRecords()
+    const copyRecords = openedExerciseRecords.recordSets.slice()
+    return copyRecords.sort(
+      (setA, setB) => setA.groupingValue - setB.groupingValue
+    )
+  }, [JSON.stringify(stateStore.openedExerciseSets)])
```

Comment:
This looks wrong.
Can't we return a computed from the memo if change detection is the issue?